frontend/src/metabase/visualizer/utils/update-viz-settings-with-refs.ts

```diff
@@ -0,0 +1,51 @@
+import type { VisualizationSettings } from "metabase-types/api";
+
+/**
+ * Recursively converts visualization settings to use the new column references.
+ *
+ * If the settings contain the color for a series called, say, `avg` (`{colors: {avg: "#000"}}`),
+ * and the column reference for `avg` is `COLUMN_1`, this function will convert it
+ * to `{colors: {COLUMN_1: "#000"}}`.
+ *
+ *
+ * @param settings the settings to convert
+ * @param columnsToRefs the mapping of column names to their references
+ * @returns the converted settings
+ */
+export function updateVizSettingsWithRefs(
```

Comment:
Wdyt about changing this signature to `deepUpdateObjectKeys(object: unknown, mapping: Record<string, string>)` to make it more precise and remove type casts like `null as any` in specs?

Reply:
I don't want this function to be too generic, in fact I think the specs that test it against non-object values is meaningless (I'll remove it), TS would catch that.